src/Components/Dialog/__tests__/Dialog.test.js

```diff
@@ -0,0 +1,30 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Dialog } from '../Dialog';
+
+describe('Dialog', () => {
+  test('renders all details', () => {
+    const title = 'Title test';
+    const children = 'children test';
+
+    render(
+      <Dialog title={title} onClose={jest.fn()}>
+        <button>{children}</button>
+      </Dialog>
+    );
+
+    expect(screen.getByText(title)).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: children })).toBeInTheDocument();
+  });
+
+  test('handles close event with calling callback', async () => {
```

Comment:
Something like this will be more clear in my opinion:

```suggestion
  test('calls close callback on close button click', async () => {
```

or something like this.